pkg/sql/opt/optbuilder/delete.go

```diff
@@ -63,6 +63,12 @@ func (b *Builder) buildDelete(del *tree.Delete, inScope *scope) (outScope *scope
 	// Find which table we're working on, check the permissions.
 	tab, depName, alias, refColumns := b.resolveTableForMutation(del.Table, privilege.DELETE)
 
+	if tab.IsVirtualTable() {
```

Comment:
"delete from view" seems like a bit of a misleading error message. any reason why we don't say "table" or "virtual table"?

Reply:
This was based on the postgres message for similar cases:
```
postgres=# delete from pg_cursors;
ERROR:  cannot delete from view "pg_cursors"
DETAIL:  Views that do not select from a single table or view are not automatically updatable.
HINT:  To enable deleting from the view, provide an INSTEAD OF DELETE trigger or an unconditional ON DELETE DO INSTEAD rule.
```

Reply:
It's worth mentioning that we don't expect to get to this point, since virtual tables shouldn't allow mutation access privileges anyway. This is a defensive check to prevent an internal error we saw in a sentry report, but don't know how to reproduce.

Reply:
i see, thanks for explaining